.github/workflows/compiler-zoo.yml

```diff
@@ -16,33 +16,72 @@ jobs:
       matrix:
         zoo: [
           {
-            cc: gcc-7
+            cc: gcc-7,
+            distro: ubuntu-latest
           }, {
-            cc: gcc-8
+            cc: gcc-8,
+            distro: ubuntu-latest
           }, {
-            cc: gcc-9
+            cc: gcc-9,
+            distro: ubuntu-latest
           }, {
-            cc: gcc-10
+            cc: gcc-10,
+            distro: ubuntu-latest
           }, {
-            cc: clang-6.0
+            cc: gcc-11,
+            distro: ubuntu-22.04
           }, {
-            cc: clang-7
+            cc: gcc-12,
+            distro: ubuntu-22.04
           }, {
-            cc: clang-8
+            cc: clang-6.0,
+            distro: ubuntu-latest
           }, {
-            cc: clang-9
+            cc: clang-7,
+            distro: ubuntu-latest
           }, {
-            cc: clang-10
+            cc: clang-8,
+            distro: ubuntu-latest
           }, {
-            cc: clang-11
+            cc: clang-9,
+            distro: ubuntu-latest
           }, {
-            cc: clang-12
+            cc: clang-10,
+            distro: ubuntu-latest
+          }, {
+            cc: clang-11,
+            distro: ubuntu-latest
+          }, {
+            cc: clang-12,
+            distro: ubuntu-latest
+          }, {
+            cc: clang-13,
+            distro: ubuntu-22.04
+          }, {
+            cc: clang-14,
+            distro: ubuntu-22.04
           }
         ]
-    runs-on: ubuntu-latest
+    # Not ubuntu-latest in order to get newer GCCs + Clang; can switch back once 'latest' is updated
+    runs-on: ${{ matrix.zoo.distro }}
     steps:
     - name: install packages
       run: |
+        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+
+        echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy main" | sudo tee /etc/apt/sources.list.d/llvm.list
+        echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy main" | sudo tee /etc/apt/sources.list.d/llvm.list
+        echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-13 main" | sudo tee /etc/apt/sources.list.d/llvm.list
+        echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-13 main" | sudo tee /etc/apt/sources.list.d/llvm.list
+        echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-14 main" | sudo tee /etc/apt/sources.list.d/llvm.list
+        echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-14 main" | sudo tee /etc/apt/sources.list.d/llvm.list
+
+        echo "deb https://ppa.launchpadcontent.net/ubuntu-toolchain-r/ppa/ubuntu jammy main" | sudo tee /etc/apt/sources.list.d/gcc.list
+        echo "deb-src https://ppa.launchpadcontent.net/ubuntu-toolchain-r/ppa/ubuntu jammy main" | sudo tee /etc/apt/sources.list.d/gcc.list
+
+        # From https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/ppa
+        sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 60C317803A41BA51845E371A1E9377A2BA9EF27F
+
```

Comment:
Why is this needed? Or, for which clang/gcc versions this is needed?

Reply:
Checked again and it's a left over experiment from trying to make it all work with Ubuntu 20.04. Dropped as nothing needs it now.

In theory, we could keep the Clang one to get latest 13.x and 14.x, but let's not worry about it for now.